### PR TITLE
Add reporting for release CodeBuild jobs

### PR DESCRIPTION
**Pull request by dagnir**
## Motivation and Context
Easier and faster identification of failed tests in builds.

## Description
Trawling through the logs looking for the tests that failed is time consuming,
especially because tests are run in parallel, making the logs harder to follow.
This uses the test result dashboarding feature from CodeBuild to make it much
easier and faster to find tests that failed during a release.

## Testing

Tested the buildspecs in personal account using the SDK v2.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the **CONTRIBUTING** document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: buildspecs/build.yml [hunk 1]
```diff
@@ -15,4 +15,11 @@ phases:
           cd test/module-path-tests
           mvn package
           mvn exec:exec -P mock-tests
-        fi
\ No newline at end of file
+        fi
+    finally:
```
Comment (millems) [inline buildspecs/build.yml:19]: TIL about "finally"

File: buildspecs/build.yml [hunk 2]
```diff
@@ -15,4 +15,11 @@ phases:
...
+      - mkdir -p codebuild-test-reports
+      - find ./ -name 'TEST-*.xml' -type f -exec cp {} codebuild-test-reports/ \;
```
Comment (millems) [inline buildspecs/build.yml:21]: Isn't there HTML output as well?

Comment (dagnir) [reply buildspecs/build.yml:21]: The reports only process the XML files

File: buildspecs/build.yml (hunk 1 above)
Comment (dagnir) [reply buildspecs/build.yml:19]: same!

File: buildspecs/build.yml (hunk 2 above)
Comment (millems) [reply buildspecs/build.yml:21]: Ah, cool, I didn't know that they generated dashboards.

**Review by millems — APPROVED** [review]

**Pull request merged**